src/gwt/src/org/rstudio/studio/client/common/shell/ShellWidget.java

```diff
@@ -433,7 +435,15 @@ public void consolePrompt(String prompt, boolean showInput)
    @Override
    public void ensureInputVisible()
    {
-      scrollIntoView();
+      // NOTE: we don't scroll immediately as this is normally called
+      // in response to mutations of the console input buffer, and so
+      // we need to wait until Ace has finished rendering in response
+      // to that change.
+      scrollIntoViewPending_ = true;
+      
+      // In case there wasn't an Ace render in-flight, force a check
+      // for pending scroll (which will then force the cursor into view)
+      Scheduler.get().scheduleDeferred(() -> checkForPendingScroll());
```

Comment:
Should we skip doing this if `scrollIntoViewPending_` is already true so there's no way for requests to stack up?

Reply:
Good idea!